### ISSUE TITLE
redirect->route wrong docu or default values

### ISSUE THREAD

**Issue by puschie286** [bcit-ci/CodeIgniter4#1125]
hey, currently the redirect function not work like the docu describe it. [Link](https://bcit-ci.github.io/CodeIgniter4/general/common_functions.html#redirect)
it result in an
`Argument 3 passed to CodeIgniter\HTTP\Response::redirect() must be of the type integer, null given`
because route has null as default $code value which is not valid ( but suggested by the documentation )

so what behavior is the right one ?
(personally i would remove the default value of $code ( and switch $params and $code ) )
`public function route(string $route, array $params=[], int $code=null, string $method='auto')`

**Comment by lonnieezell** [comment]
Please provide a code sample how you're using this. The docs you linked to are for the common function `redirect()` while it looks like samples you're showing are calling it directly on the Response object, so I'm a bit confused.

**Comment by puschie286** [comment]
```
// Go to a named/reverse-routed URI
return redirect()->route('named_route');
```

**Comment by puschie286** [comment]
thats not a sample -> its the route function definition. just to make it easier to understand the problem ( PS: it should show that the default parameter are invalid ( and they are used by the redirect function ) )

**Comment by puschie286** [comment]
fixed by #1147
